homeassistant/components/switch/recswitch.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Ankuoo RecSwitch MS6126 devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.recswitch/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.recswitch import DATA_RSH
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['recswitch']
+
+DEFAULT_NAME = 'RecSwitch {0}'
+
+POLLING_INTERVAL = timedelta(seconds=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Setup the device and register on recswitch network."""
+    ip_address = config.get(CONF_HOST)
+    mac_address = config.get(CONF_MAC).upper()
+    device_name = config.get(CONF_NAME)
+
+    device = hass.data[DATA_RSH].register_device(mac_address, ip_address)
+    async_add_entities([RecSwitchSwitch(device, device_name, mac_address)])
+
+
+class RecSwitchSwitch(SwitchDevice):
+    """Representation of a recswitch device."""
+
+    def __init__(self, device, device_name, mac_address):
+        self.gpio_state = self.assumed_state
+        self.device = device
+        self.device_name = device_name
+        self.device.report_gpio_change = self.report_gpio_change
+        if not self.device_name:
+            self.device_name = DEFAULT_NAME.format(mac_address)
+
+    @property
+    def name(self):
+        return self.device_name
+
+    @property
+    def assumed_state(self):
+        return False
+
+    @property
+    def should_poll(self):
+        return True
+
+    @property
+    def is_on(self):
+        return self.gpio_state
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on the switch."""
+        ret = await self.device.set_gpio_status(True)
+        self.gpio_state = ret.state
+
+    async def async_turn_off(self, **kwargs):
+        """Turn off the switch."""
+        ret = await self.device.set_gpio_status(False)
+        self.gpio_state = ret.state
+
+    async def async_update(self):
+        """Get the current switch status."""
+        ret = await self.device.get_gpio_status()
+        self.gpio_state = ret.state
+
+    async def report_gpio_change(self, ret):
+        """Callback on switch status change."""
+        self.gpio_state = ret.state
```

Comment:
Home assistant state needs to be scheduled to be updated too after this. Use `self.hass.async_schedule_update_ha_state`.